src/components/Form.tsx

```diff
@@ -12,10 +12,12 @@ export interface FormProps {
   buttonContent?: string | JSX.Element;
   buttonClass?: string;
   className?: string;
-  errorText?: string;
-  successText?: string;
-  infoText?: string;
+  errorText?: string | JSX.Element | Array<JSX.Element>;
+  successText?: string | JSX.Element | Array<JSX.Element>;
+  infoText?: string | JSX.Element | Array<JSX.Element>;
+  loadingText?: string | JSX.Element | Array<JSX.Element>;
   disableButton?: boolean;
+  withoutButton?: boolean;
```

Comment:
I'm not sure you want to have a form without a button. The checkbox example you gave is okay but from the toggle article you sent, the toggle makes more sense to have there instead of a checkbox. The checkbox doesn't give instant results as the toggle does. But in general, I would not think a form should not have a button (in the no-javascript state the form will not work without a button).

Reply:
I implemented that for the sake of one particular use case: the nested rights form on the cover editor page in circ-web, which has two input fields but no button (you submit it via the parent form, which is disabled until conditions from the other nested form have been met).  But I'm not actually convinced that that's how the rights form should work anyway (I think you should be able to update the explanation of rights without having to first mess around with the cover image); that's on my list of issues with the cover editor form to look into.  Thoughts?

Anyway...I agree that a toggle would make more sense than a checkbox in the example: with the toggle, the no-JS thing will not be an issue, because the toggle IS a button.  I think by far the most reasonable use case for using `withoutButton` would be something like that, actually: it's not that you're creating a form that doesn't have a button; it's just that you want to use some special type of button (by passing it into the `content` prop), rather than having the Form automatically generate a Button component for you.  I'll see what I can do to make that clearer.